src/connectiondataholder.cpp

```diff
@@ -224,14 +232,27 @@ void ConnectionDataHolder::updateIpAddress() {
   connect(request, &NetworkRequest::requestCompleted,
           [this](QNetworkReply*, const QByteArray& data) {
             logger.log() << "IP address request completed";
-
             QJsonDocument json = QJsonDocument::fromJson(data);
             if (json.isObject()) {
               QJsonObject obj = json.object();
 
+              QJsonValue country = obj.take("country");
+              if (m_checkStatusTimer.isActive() &&
+                  country.toString().toLower() !=
+                      MozillaVPN::instance()->currentServer()->countryCode()) {
+                // In case the country-we're reported in does not match the
+                // connected server we may retry only once.
+                logger.log() << "Reported ip not in the right country, retry!";
+                TimerSingleShot::create(this, 3000, [this]() {
+                  m_updatingIpAddress = false;
```

Comment:
This line can be removed. If there is another check in progress, we abort the new one.